docs/preview/features/message-pumps/customization.md

```diff
@@ -44,6 +46,43 @@ public class OrderMessagePump : MessagePump
 }
 ```
 
+### Delegating deserialization
+
+You can also choose to delegate the message deserialization to different deserializers. 
+This allows for more reuse, asynchronous, and is available without altering your message pump. 
+
+You start by implemeting an `IMessageBodyHandler`. The following example shows how an expected type can be transformed to something else. 
+The result type (in this case `OrderBatch`) will be then be used to check if there is an `IMessageHandler` registered with that message type.
+
+```csharp
+public class OrderBatchMessageBodyHandler : IMessageBodyHandler
+{
+    public async Task<MessageResult> DeserializeMessageAsync(string messageBody)
+    {
+        var orders = JsonConvert.DeserializeObject<Order[]>(messageBody);
+        return MessageResult.Success(new OrderBatch(orders));
+    }
+}
+```
+
+The registration of these message body handlers can be done just as easily as an `IMessageHandler`:
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
```

Comment:
Shouldn't the registration be linked to the message pump that is being registered? Either generally or per message handler because otherwise we will do a lot of deserialization to match things, no?

```csharp
services.AddServiceBusQueueMessagePump("ARCUS_SERVICEBUS_ORDERS_CONNECTIONSTRING")
            .WithServiceBusMessageHandler<OrdersMessageHandler, Order, OrderBatchMessageBodySerializer>();
```

I'm just thinking out loud.

Reply:
We could do that, yeah, we should then register the message handler differently but it's possible.

Reply:
Should I move to this approach?

Reply:
Moved to this approach, resolving.